diff --git a/obfuscate.h b/obfuscate.h
index aa2480a..0c46741 100644
--- a/obfuscate.h
+++ b/obfuscate.h
@@ -228,6 +228,7 @@ namespace ay
 // ay::obfuscated_data object with global lifetime that has functions for
 // decrypting the string and is also implicitly convertable to a char*
 #define AY_OBFUSCATE(data) AY_OBFUSCATE_KEY(data, AY_OBFUSCATE_DEFAULT_KEY)
+#define _O(data) AY_OBFUSCATE(data)
 
 // Obfuscates the string 'data' with 'key' at compile-time and returns a
 // reference to a ay::obfuscated_data object with global lifetime that has
